### PR TITLE
UT test case to check device and group based on name and type entity

### PR DESCRIPTION
**Pull request by MayankSingh-NEC**
This PR fixes the issue: https://github.com/telefonicaid/iotagent-node-lib/issues/1213

### CODE REVIEW

**Comment by fgalan** [comment]
Thanks for your contribution!

Note that the lint step is reporting a failure:

```
 /home/runner/work/iotagent-node-lib/iotagent-node-lib/test/unit/mongodb/mongodb-registry-test.js
Error:   528:45  error  Unexpected string concatenation of literals  no-useless-concat
```

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,3 +1,4 @@
 - Add: NGSI-LD Unsupported Endpoints return valid error responses (#1277)
 - Fix: protect explicitAttrs evaluation of a bad formed expression (#1175)
 - Fix: ensure GET /ngsi-ld/v1/entities/<entity-id> returns a valid response (#1276)
+- Add: UT test case to check device and group based on name and type entity (#1279)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:4]:
```suggestion
```

Changelog is about bugfixing or functional changes. New tests of course are a good adition, but they are a "hardening" work that is not included in the changelog.

**Review by AlvaroVega — COMMENTED** [review]
Thanks a lot for your contribution @MayankSingh-NEC

**Comment by fgalan** [comment]
Thanks for the contribution!

However, the issue title is

> Add test to check entity comand get propers device and group based on name and type entity

But is seems the test included in this PR are not related with commands...

**Comment by MayankSingh-NEC** [comment]
> Thanks for the contribution!
> 
> However, the issue title is
> 
> > Add test to check entity comand get propers device and group based on name and type entity
> 
> But is seems the test included in this PR are not related with commands...

Hi @fgalan , Thanks for reviewing the PR. Do I need to write the UT test case for the entity command specifically for northbound/Context-server-ngsi or anything else?

Also, the UT test case that I've written for "deviceRegisteryMemory" and "deviceRegistryMongoDB" are not available in the existing UT. Do we need the test cases for the deviceRegisteryMemory and deviceRegistryMongoDB files? Please let me your opinion. Thanks.

**Comment by AlvaroVega** [comment]
Yeah, it could be great an UT test case for the entity command specifically for northbound/Context-server-ngsi

**Comment by fgalan** [comment]
@MayankSingh-NEC the tests that @AlvaroVega mentions should be included as a minimum.

However, any other test beyond that is also welcome :) Thus, the current tests in the PR are ok and any other addition that makes sense to include.

**Comment by MayankSingh-NEC** [comment]
> @MayankSingh-NEC the tests that @AlvaroVega mentions should be included as a minimum.
> 
> However, any other test beyond that is also welcome :) Thus, the current tests in the PR are ok and any other addition that makes sense to include.

Thank you @AlvaroVega and @fgalan ! Recently, I'm working on an another issue. If the above PR's test cases are fine to merge. Then, it would be grateful for me. Thanks.
Also, could you let me know more about the title description*

**Comment by fgalan** [comment]
>  Also, could you let me know more about the title description*

Not sure what do you mean by "title description". Do you mean the title of the PR (i.e. "
UT test case to check device and group based on name and type entity")? Could you clarify please?

**Comment by MayankSingh-NEC** [comment]
> > Also, could you let me know more about the title description*
> 
> Not sure what do you mean by "title description". Do you mean the title of the PR (i.e. "UT test case to check device and group based on name and type entity")? Could you clarify please?

Hi @fgalan , Actually, I was referring the "title description" as the description of issue #1213 (i.e. "Add test to check entity command get propers device and group based on name and type entity").

**Comment by fgalan** [comment]
Let's do the following:

* Merge this PR, as it includes new unit tests and new unit tests are good in general (they enhance regression :)
* Please feel free to contribute with new unit tests (the ones requested by #1213 or others) in new PRs
* If you have any doubt or question regarding issue #1213 title, please ask in that issue and we will continue discussing there

Thanks!

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**